bainu_custom/report/invoice_reports.xml

```diff
@@ -0,0 +1,172 @@
+                        <div class="col-xs-2" t-if="o.partner_id.ref">
+                            <strong>Customer Code:</strong>
+                            <p t-field="o.partner_id.ref"/>
+                        </div>
+                        <div class="col-xs-2" t-if="o.reference">
+                            <strong>Reference:</strong>
+                            <p t-field="o.reference"/>
+                        </div>
+                    </div>
+                    <table class="table table-condensed">
+                        <thead>
+                            <tr>
+                                <th>Description</th>
+                                <th>Quantity</th>
+                                <th class="text-right">Unit Price</th>
+                                <th class="text-right"></th>
+                                <th class="text-right">Amount</th>
+                            </tr>
+                        </thead>
+                        <tbody class="invoice_tbody">
+                            <tr t-foreach="o.invoice_line" t-as="l">
+                                <td><span t-field="l.name"/></td>
+                                <td>
+                                    <span t-field="l.quantity"/>
+                                    <span t-field="l.uos_id"  groups="product.group_uom"/>
+                                </td>
+                                <td class="text-right">
+                                    <span t-field="l.price_unit"/>
+                                </td>
+                                <td class="text-right" groups="sale.group_discount_per_so_line">
+                                    <t t-if="l.discount">
+                                        <span t-field="l.discount"/> % Dto.
+                                    </t>
+                                </td>
+                                <td class="text-right">
+                                    <span t-field="l.price_subtotal" 
+                                        t-field-options='{"widget": "monetary", "display_currency": "o.currency_id"}'/>
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                    <div class="row" t-if="o.tax_line">
+                        <div class="col-xs-6">
+                            <table class="table table-condensed">
+                                <thead>
+                                    <tr>
+                                        <th>Tax</th>
+                                        <th class="text-right">Base</th>
+                                        <th class="text-right">Amount</th>
+                                    </tr>
+                                </thead>
+                                <tbody>
+                                    <tr t-foreach="o.tax_line" t-as="t">
+                                        <td><span t-field="t.tax_text"/></td>
```

Comment:
Creo que si pones `<span t-esc="t.name.split()[0]"/> ` no tienes que crear ni campo ni código ni nada, y la solicitud podrías resolverlo solo modificando el informe.
Por otro lado, si se cambian los nombres de los impuestos, por ejemplo que uno se llame IVA 21% en vez de cogerte 21% te cogería IVA. Si fuese Recargo de equivalencia, te pondría Recargo. Yo lo haría de la siguiente forma para contemplar esos casos y para que si hay un % te coja el 21% por ejemplo, y en el caso de que no haya % en el nombre del impuesto te muestre el nombre completo.
`<t t-if="'%' in t.name"><span t-esc="t.name.split('%')[0].split()[-1:]"/>%</t><t t-if="'%' not in t.name"><span t-esc="t.name"/></t>`